### PR TITLE
Add use cases for budget management

### PR DESCRIPTION
**Pull request by woojiahao**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -305,6 +305,57 @@ the filtered transaction with a success message.
...
+3. UniCa$h sets the global budget.
+4. UniCa$h displays the new budget.
+
+   Use Case ends.
+
+**Extensions:**
+
+- 1a. User enters the incorrect format (missing fields)
+  - 1a1. UniCa$h displays an error message with the correct command format
+  - Use case resumes at step 1.
```
Comment (lipwei1808) [inline docs/DeveloperGuide.md:322]: Should the entering incorrect format fall under 2a. (Referring to how the other Use cases are done above)

**Review by lipwei1808 — COMMENTED** [review]
Just one small question on your first Use case

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (woojiahao) [reply docs/DeveloperGuide.md:322]: I think it doesn't make sense for it to start with 2a if step 1 was already that the user entered the command in the right format

Comment (lipwei1808) [reply docs/DeveloperGuide.md:322]: That makes sense, we should probably update the rest too.

**Review by lipwei1808 — APPROVED** [review]

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (woojiahao) [reply docs/DeveloperGuide.md:322]: Yup, I can make a PR o pick this up

**Pull request merged**